frontend/src/app/modules/user/user-profile/user-profile.component.ts

```diff
@@ -103,4 +118,52 @@ export class UserProfileComponent extends BaseComponent implements OnInit {
             this.toastrNotification.showError('User not found');
         }
     }
+
+    private getUserFriendships() {
+        this.userService
+            .getUserFriendships(this.currentUser.id)
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe({
+                next: (userFriendsInfo) => {
+                    this.userFriendsIds = userFriendsInfo.friendships.map((f) => f.friendId);
+                    this.currentUser.friendships = userFriendsInfo.friendships;
+                    this.friendshipId = this.currentUser.friendships.find((f) => f.friendId === this.user.id)?.friendshipId;
```

Comment:
why do we need all user friendships in frontend?

Reply:
I just used the existing methods to work with friends, but if it's an issue I'll create new ones for this cause

Reply:
it would be better to create a separate method for only this purpose